pontoon/base/utils.py

```diff
@@ -374,10 +378,25 @@ def handle_upload_content(slug, code, part, f, user):
             )
         )
     )
+    log.info(
+        f"Sample keys in entities_qs: {[entity.key for entity in entities_qs[:5]]}"
+    )
+
+    # Log the size of entities_qs before entities_dict is created
+    log.info(f"Size of entities_qs: {len(entities_qs)}")
+
     entities_dict = {entity.key: entity for entity in entities_qs}
 
+    log.info(f"Size of entities_dict: {len(entities_dict)}")
+
+    # Log a few sample key-value pairs from entities_dict
+    sample_items = list(entities_dict.items())[:5]  # Get first 5 items
+    for key, value in sample_items:
+        log.info(f"Sample key-value pair: {key}: {value}")
+
     for vcs_translation in resource_file.translations:
         key = vcs_translation.key
+        log.info(f"Processing translation - Key: {key}, Translation: {vcs_translation}")
         if key in entities_dict:
```

Comment:
Seems like there is a mismatch between the key formats in `key`:  (e.g. `["notificationContent", "description"]`) and `entities_dict` (e.g. `"notificationContent"`).